mne/preprocessing/ica.py

```diff
@@ -962,6 +954,145 @@ def get_components(self):
         return np.dot(self.mixing_matrix_[:, :self.n_components_].T,
                       self.pca_components_[:self.n_components_]).T
 
+    def get_explained_variance_ratio(
+        self, inst, *, components=None, ch_type=None
+    ):
+        """Get the proportion of data variance explained by ICA components.
+
+        Parameters
+        ----------
+        inst : mne.io.BaseRaw | mne.BaseEpochs | mne.Evoked
+            The uncleaned data.
+        components : array-like of int | int | None
+            The component(s) for which to do the calculation. If more than one
+            component is specified, explained variance will be calculated
+            jointly across all supplied components. If ``None`` (default), uses
+            all available components.
+        ch_type : 'mag' | 'grad' | 'planar1' | 'planar2' | 'eeg' | array-like of str | None
+            The channel type(s) to include in the calculation. If ``None``, all
+            available channel types will be used. Note that the value of this
+            parameter may change the return type (float or dictionary).
+
+        Returns
+        -------
+        float | dict [str, float]
+            The fraction of variance in ``inst`` that can be explained by the
+            ICA components. If only a single ``ch_type`` was given, a float
+            will be returned. Otherwise, a dictionary with channel types as
+            keys and explained variance ratios as values.
```

Comment:
honestly I would always return a dict as otherwise we'll have for example to branch in the pipeline code etc. wdyt?

Reply:
I was a bit hesitant to do that because if you have for example only EEG data, you'd need to do:
```python
var_explained = ica.get_explained_variance_ratio(inst)['eeg']
```
which I felt was kind of odd … but you're of course right, always returning a dict would make things more consistent, so that's probably better.

@drammock @cbrnr WDYT?

Reply:
@agramfort I've implemented your suggestion, and we're always returning a `dict` now